### PR TITLE
kvutils: Create our `Identifier` message and remove references to Daml-LF one [KVL-1167]

### PR DESCRIPTION
**Pull request by hubert-da**
### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueCommitting.scala [hunk 1]
```diff
@@ -276,20 +274,4 @@ object KeyValueCommitting {
     }
     outputs.result()
   }
-
-  private def contractKeyToStateKey(
-      key: GlobalKey
-  ): DamlStateKey = {
-    // NOTE(JM): The deserialization of the values is meant to be temporary. With the removal of relative
```
Comment (hubert-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueCommitting.scala:283]: This comment is no longer valid (see https://github.com/digital-asset/daml/pull/8519/files#diff-e355a2d6edc45d41fc83bf220e75220dc4d0873cd0d2641e9378eabb5535c368R260)

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/batch/ConflictDetection.scala [hunk 2]
```diff
@@ -104,8 +105,7 @@ class ConflictDetection(val damlMetrics: Metrics) {
           // NOTE(JM): We show the template id as the other piece of data we have is the contract key
           // hash, which isn't very useful as it's an implementation detail not exposed over ledger-api.
           val templateId =
-            ValueCoder
-              .decodeIdentifier(key.getContractKey.getTemplateId)
+            Try(Conversions.decodeIdentifier(key.getContractKey.getTemplateId))
```
Comment (nicu-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/batch/ConflictDetection.scala:108]: Do we want to just swallow this exception?

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala [hunk 3]
```diff
@@ -101,8 +103,40 @@ object Conversions {
...
+      name <- DottedName
+        .fromSegments(nameSegments)
+        .left
+        .map(_ => Err.DecodeError("Identifier", s"Invalid name segments: $nameSegments"))
+    } yield Ref.Identifier(pkgId, QualifiedName(module, name))
+
+    errorOrRefId match {
+      case Right(refId) => refId
+      case Left(decodeError) => throw decodeError
+    }
```
Comment (nicu-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala:138]: Any specific reason why we get rid of the Either here and not just return the Either?

Comment (hubert-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala:138]: This is the convention in `kvutils`/`Conversions` but there is just one usage here, so I have no strong opinion on this.

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/batch/ConflictDetection.scala (hunk 2 above)
Comment (hubert-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/batch/ConflictDetection.scala:108]: Yes, this was the case previously as well, as this is just for logging.

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala (hunk 3 above)
Comment (hubert-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala:138]: Actually, let me change this.

**Review by SamirTalwar-DA — APPROVED** [review]

**Review by nicu-da — APPROVED** [review]

**Pull request merged**